contributors/design-proposals/autoscaling/hpa-step-scaling.md

```diff
@@ -0,0 +1,143 @@
+# HPA: Threshold Based Step Scaling Policy Proposal
+
+## Target of HPA
+
+The target of HPA is to dynamically adjusts the amount of allocated resources (pods) to meet changes in
+workload demands. Basically, we evaluate a good HPA scaling policy as:
+
+* Can achieve high resource utilization
+* Responsive: scale to the desire number of replicas fast
+
+## Problem
+
+The Horizontal Pod Autoscaler (HPA) is implemented as a Kubernetes API resource and a controller.
+Current HPA only implement “target tracking based scaling policy”: the controller periodically adjusts 
+the number of replicas in a replication controller or deployment to match the observed average CPU 
+utilization to the target specified by user.
+
+In a nutshell, the target number of pods is calculated from the following formula:
+```
+TargetNumOfPods = ceil(sum(CurrentUtilization) / TargetUtilization)
```

Comment:
this is wrong -- it's `avg(CurrentUtilization) / TargetUtilization`, not `sum(CurrentUtilization) / TargetUtilization`